src/_pytest/pytester.py

```diff
@@ -1396,16 +1400,19 @@ def _no_match_line(self, pat, match_func, match_nickname):
         """
         __tracebackhide__ = True
         nomatch_printed = False
+        wnick = len(match_nickname) + 1
         try:
             for line in self.lines:
                 if match_func(line, pat):
                     self._log("%s:" % match_nickname, repr(pat))
-                    self._log("   with:", repr(line))
-                    pytest.fail(self._log_text)
+                    self._log("{:>{width}}".format("with:", width=wnick), repr(line))
+                    pytest.fail(self._log_text.lstrip())
```

Comment:
Note that the lstrip is kind of a hack, to not have leading whitespace with the pytest.fail.Exception message.
Could be revisited after #6003 .